### PR TITLE
Feature/rework issue 297

### PR DESCRIPTION
**Pull request by FranFin**
Changes implemented:

- Added "Integration into External Platforms" page in Overview section
- Added STM32CubeMX to external build systems in Concepts section
- Changed boards organization in Supprted HW page in Overview page and add transport/RTOS/external BS info
- Removed Transports page and absorb info into HW page
- Reorganized tutorial sections: Core -> First-steps; NuttX-specific -> Unmaintained. Also, tutorials have been moved from one section to another.
- Update Users, collaborators etc section w/ new user: Hydra System

See https://github.com/micro-ROS/micro-ROS.github.io/issues/297 for ref.

### CODE REVIEW

File: _docs/concepts/build_system/external_build_systems/index.md [hunk 1]
```diff
@@ -22,3 +22,9 @@ The procedure for configuring the built micro-ROS library is based in `colcon.me
 The [micro-ROS for Arduino](https://github.com/micro-ROS/micro_ros_arduino) support package is a special port of micro-ROS provided as a set of precompiled libraries for specific platforms. The main reason for this approach is that Arduino does not allow the build of a complex library such as micro-ROS, so by using this approach a ready-to-use solution is provided to the Arduino users.
 
 Along with this support package, there are [detailed instructions](https://github.com/micro-ROS/micro_ros_arduino#how-to-build-the-precompiled-library) for rebuilding the micro-ROS for Arduino libraries for users that need to tune the default configuration.
+
+## micro-ROS for STM32CubeMX
+
+The [micro-ROS for STM32CubeMX](https://github.com/micro-ROS/micro_ros_stm32cubemx_utils) package is a set of utilities which enables the seamless configuration, set-up and integration of micro-ROS into any STM32 controller based project. As such, it allows micro-ROS to be virtually supported by the full set of boards offered by <a href="https://www.st.com/content/st_com/en.html">STMicroelectronics</a>.
```
Comment (pablogs9) [inline _docs/concepts/build_system/external_build_systems/index.md:28]:
```suggestion
The [micro-ROS for STM32CubeMX](https://github.com/micro-ROS/micro_ros_stm32cubemx_utils) package is a set of utilities which enables the seamless configuration, set-up and integration of micro-ROS into an STM32 controller based project. As such, it allows micro-ROS to be virtually supported by the full set of boards offered by <a href="https://www.st.com/content/st_com/en.html">STMicroelectronics</a>.
```

File: _docs/concepts/build_system/external_build_systems/index.md [hunk 2]
```diff
@@ -22,3 +22,9 @@ The procedure for configuring the built micro-ROS library is based in `colcon.me
...
+
+Its usage is based on Dockers, via a prepared [Dockerfile](https://github.com/micro-ROS/micro_ros_stm32cubemx_utils/blob/foxy/microros_component/Dockerfile) which eases the project configuration and execution.
```
Comment (pablogs9) [inline _docs/concepts/build_system/external_build_systems/index.md:30]:
```suggestion
Its usage is based on Dockers, via a prepared [Dockerfile](https://github.com/micro-ROS/micro_ros_stm32cubemx_utils/blob/foxy/microros_component/Dockerfile) which eases micro-ROS library generation outside of a ROS 2 environment.
```

File: _docs/overview/hardware/index.md [hunk 3]
```diff
@@ -34,63 +34,39 @@ The main targets of micro-ROS are mid-range 32-bits microcontroller families. Us
...
+        &#127758; <b>Resources:</b>
         <ul>
             <li><a href="https://www.espressif.com/en/products/socs/esp32">Official website</a></li>
             <li><a href="https://www.espressif.com/sites/default/files/documentation/esp32-wroom-32e_esp32-wroom-32ue_datasheet_en.pdf">Datasheet</a></li>
-        </ul>    
+        </ul>
+        &#9881; <b>Supported platforms:</b>
+        <ul>
+            <li><b>RTOS:</b> FreeRTOS</li>
+            <li><b>External platform:</b> ESP-IDF</li>
```
Comment (pablogs9) [inline _docs/overview/hardware/index.md:66]: Should we add links here to the micro-ROS component for the platform if exists?

**Comment by pablogs9** [comment]
In general I think that we should discuss the concept of "External platform" because sometimes we are talking about a toolchain, other to a development environment, other a build system...

File: _docs/overview/hardware/index.md (hunk 3 above)
Comment (FranFin) [reply _docs/overview/hardware/index.md:66]: Yes. I'll do it right away.

**Comment by FranFin** [comment]
@pablogs9 I agree. What do you think about adding this kind of discussion at the beginning of the newly added page?

**Comment by pablogs9** [comment]
So I propose to change all `External Platforms` with `External Tools`

**Review by ralph-lange — CHANGES_REQUESTED** [review]
Nice page! This will help a lot to understand the different build systems.

**Review by pablogs9 — DISMISSED** [review]

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**